service/src/main/java/bio/terra/externalcreds/controllers/OauthApiController.java

```diff
@@ -0,0 +1,37 @@
+package bio.terra.externalcreds.controllers;
+
+import bio.terra.externalcreds.generated.api.OauthApi;
+import bio.terra.externalcreds.generated.model.Provider;
+import bio.terra.externalcreds.services.*;
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import jakarta.servlet.http.HttpServletRequest;
+import lombok.SneakyThrows;
+import org.springframework.http.ResponseEntity;
+import org.springframework.stereotype.Controller;
+
+@Controller
+public record OauthApiController(
+    HttpServletRequest request,
+    ObjectMapper mapper,
+    ProviderService providerService,
+    ExternalCredsSamUserFactory samUserFactory)
+    implements OauthApi {
+
+  @Override
+  public ResponseEntity<String> getAuthUrl(Provider providerName, String redirectUri) {
```

Comment:
This method name will have to change if the `operationId` in the swagger definition changes